seen_blocks.go

```diff
@@ -22,13 +22,27 @@ import (
 )
 
 type SeenBlockCache struct {
-	M           map[string]bool
-	filename    string
-	keepSize    uint64
+	M        map[string]bool
+	filename string // optional, in-memory mode if not set
+	keepSize uint64
```

Comment:
was confused by the name, this is sort of a `maxSize` for the number of blocks in the cache?

Reply:
ok `M` is a map of seen one-block-files, by name, not a block ID, right ?

and `HighestSeen`/`LowestSeen` is its `highBlockNum` and `lowBlockNum`

Reply:
renamed !